src/launcher/java/org/truffleruby/launcher/Launcher.java

```diff
@@ -112,15 +112,19 @@ public static void main(boolean isGraal, String[] args) throws Exception {
                     break;
             }
 
-            try (Context context = createContext(Context.newBuilder(), config)) {
-                printTruffleTimeMetric("before-run");
-                final Source source = Source.newBuilder(
-                        LANGUAGE_ID,
-                        // language=ruby
-                        "Truffle::Boot.main",
-                        BOOT_SOURCE_NAME).build();
-                exitCode = context.eval(source).asInt();
-                printTruffleTimeMetric("after-run");
+            if (config.getOption(OptionsCatalog.EXECUTION_ACTION) == ExecutionAction.NONE) {
+                exitCode = 0;
+            } else {
```

Comment:
The code in MainNode could be simplified since it does not need to handle `ExecutionAction.NONE` anymore.